### PR TITLE
[chore] Bump to 3.4.0-rc.1 - use dapr 1.14.0-rc.x

### PR DESCRIPTION
**Pull request by DeepanshuA**
# Description

Using dapr/dapr 1.14.0-rc.x version.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [ ] Extended the documentation

### CODE REVIEW

**Comment by mikeee** [comment]
@DeepanshuA should have been fixed with rc.6 / https://github.com/dapr/dapr/pull/7940
Would it be possible to bump and get this merged?

File: .github/workflows/test-e2e.yml [hunk 1]
```diff
@@ -34,9 +34,9 @@ jobs:
   test-e2e:
     runs-on: ubuntu-latest
     env:
-      GOVER: 1.21
-      DAPR_CLI_VER: 1.13.0
-      DAPR_RUNTIME_VER: 1.13.0
+      GOVER: 1.22
+      DAPR_CLI_VER: 1.14.0-rc.7
+      DAPR_RUNTIME_VER: 1.14.0-rc.5
```
Comment (mikeee) [inline .github/workflows/test-e2e.yml:39]:
```suggestion
      DAPR_CLI_VER: 1.14.0-rc.8
      DAPR_RUNTIME_VER: 1.14.0-rc.7
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dapr/dapr",
-  "version": "3.3.1",
+  "version": "3.4.0-rc.1",
```
Comment (paulyuk) [inline package.json:3]: @DeepanshuA are we sticking with 3.3.1 to go with Dapr 1.14.0, or bumping to 3.4?  I see that the key bug fix you made for health probe went into 3.3.1.

**Review by paulyuk — COMMENTED** [review]
See question on version pls.

File: package.json (hunk 2 above)
Comment (DeepanshuA) [reply package.json:3]: js-sdk ver 3.3.1 was released already. We would release 3.4.0 now.

Comment (DeepanshuA) [reply package.json:3]: Once dapr 1.14 is released finally, will do so. As of now, dapr 1.14 rc versions are out, so creating js-sdk versions in sync with that.

**Comment by DeepanshuA** [comment]
test:e2e is failing intermittenntly. I have tested on local as well, it is passing.
Here, also it passed in this particular run: https://github.com/dapr/js-sdk/actions/runs/10183072399/job/28196930203?pr=611

**Review by yaron2 — APPROVED** [review]

**Comment by marcduiker** [comment]
@holopin-bot @DeepanshuA Thanks Deepanshu!

**Pull request merged**